.github/workflows/java-ci.yml

```diff
@@ -38,4 +38,10 @@ jobs:
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
         restore-keys: ${{ runner.os }}-gradle
     - run: ./gradlew check
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: test logs
+        path: |
+          **/build/testlogs
```

Comment:
I think we just need to archive `**/build/reports`. The reports have the full stack traces and usually have any stderr/stdout for each test. That may be easier than creating a separate file since I think we already aggregate the logs there.